test/ServiceTest.cpp

```diff
@@ -0,0 +1,107 @@
+    EXPECT_EQ(request.method(), http::verb::post);
+    EXPECT_EQ(request.base()[http::field::content_type],
+              "application/sparql-query");
+    EXPECT_EQ(request.base()[http::field::accept], "text/tab-separated-values");
+
+    // Parse the query.
+    std::string_view testSparqlQuery = request.body();
+    static constexpr auto testQueryRegex = ctll::fixed_string(
+        "^\\s*SELECT\\s+(.*)\\s+WHERE\\s*\\{\\s*((.*)\\n)\\s*\\}$");
+    auto match = ctre::search<testQueryRegex>(testSparqlQuery);
+    EXPECT_TRUE(match);
+    // Why do I need `template` before `get<...>` in the following? I don't need
+    // it in other places in the code (e.g, in `util/GeoSparqlHelpers.cpp`).
+    std::string_view variables = match.template get<1>();
+    std::string_view body = match.template get<2>();
+    std::string response = absl::StrCat(
+        absl::StrJoin(absl::StrSplit(variables, " "), "\t"), "\n", body);
+    // std::cout << "Reponse (TSV): " << std::endl << response;
+    co_return co_await send(ad_utility::httpUtils::createOkResponse(
+        response, request, ad_utility::MediaType::tsv));
+  });
+  httpServer.runInOwnThread();
+  Iri serviceIri{absl::StrCat("<http://localhost:", httpServer.getPort(), ">")};
+
+  // Now perform the operation for a test query.
+  parsedQuery::Service parsedServiceClause{
+      {Variable{"?x"}, Variable{"?y"}},
+      serviceIri,
+      "",
+      "{ <x>\t<y>\n<bla>\t<bli>\n<blu>\t<bla>\n<bli>\t<blu>\n }",
+      parsedQuery::GraphPattern{}};
+  Service serviceOperation(testQec, parsedServiceClause);
+  testQec->getQueryTreeCache().clearAll();
+  std::shared_ptr<const ResultTable> result = serviceOperation.getResult();
+
+  // Check the vocabulary entries.
+  VocabIndex idxX;
+  VocabIndex idxY;
+  EXPECT_TRUE(testQec->getIndex().getVocab().getId("<x>", &idxX));
+  EXPECT_TRUE(testQec->getIndex().getVocab().getId("<y>", &idxY));
+  LocalVocabIndex idxBla = LocalVocabIndex::make(0);
+  LocalVocabIndex idxBli = LocalVocabIndex::make(1);
+  LocalVocabIndex idxBlu = LocalVocabIndex::make(2);
+  EXPECT_EQ(result->_localVocab->getWord(idxBla), "<bla>");
+  EXPECT_EQ(result->_localVocab->getWord(idxBli), "<bli>");
+  EXPECT_EQ(result->_localVocab->getWord(idxBlu), "<blu>");
+
+  // Check the result.
+  EXPECT_TRUE(result);
+  EXPECT_EQ(result->_idTable.cols(), 2);
+  EXPECT_EQ(result->_idTable.rows(), 4);
+  EXPECT_EQ(result->_idTable(0, 0), Id::makeFromVocabIndex(idxX));
+  EXPECT_EQ(result->_idTable(0, 1), Id::makeFromVocabIndex(idxY));
+  EXPECT_EQ(result->_idTable(1, 0), Id::makeFromLocalVocabIndex(idxBla));
+  EXPECT_EQ(result->_idTable(1, 1), Id::makeFromLocalVocabIndex(idxBli));
+  EXPECT_EQ(result->_idTable(2, 0), Id::makeFromLocalVocabIndex(idxBlu));
+  EXPECT_EQ(result->_idTable(2, 1), Id::makeFromLocalVocabIndex(idxBla));
+  EXPECT_EQ(result->_idTable(3, 0), Id::makeFromLocalVocabIndex(idxBli));
+  EXPECT_EQ(result->_idTable(3, 1), Id::makeFromLocalVocabIndex(idxBlu));
```

Comment:
The visual overhead here is too high to see the actual contents. Please write a checkLambda
`check(2, 0, idxBlu)`
(maybe you need two functions because of the explicit `makeFrom....` functions, but this will still be more readable.